app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb

```diff
@@ -19,5 +25,185 @@ def cluster_name_href(href)
+
+      options = {
+        :name        => nic_name,
+        :interface   => interface,
+        :network_id  => network[:id],
+        :mac_address => mac_addr,
+      }.delete_blanks
+
+      logger.info("with options: <#{options.inspect}>")
+
+      if vnic.nil?
+        vm.with_provider_object do |rhevm_vm|
+          rhevm_vm.create_nic(options)
+        end
+      else
+        vnic.apply_options!(options)
+      end
+    end
+
+    def powered_off_in_provider?(vm)
+      vm.with_provider_object(&:status)[:state] == "down"
+    end
+
+    def powered_on_in_provider?(vm)
+      vm.with_provider_object(&:status)[:state] == "up"
+    end
+
+    def vm_boot_from_cdrom(operation, name)
+      begin
+        operation.get_provider_destination.boot_from_cdrom(name)
+      rescue Ovirt::VmNotReadyToBoot
+        raise OvirtServices::VmNotReadyToBoot
+      end
+    end
+
+    def vm_boot_from_network(operation)
+      begin
+        operation.get_provider_destination.boot_from_network
+      rescue Ovirt::VmNotReadyToBoot
+        raise OvirtServices::VmNotReadyToBoot
+      end
+    end
+
+    def get_template_proxy(template, connection)
+      connection ||= ext_management_system.connect
+      template_proxy = connection.get_resource_by_ems_ref(template.ems_ref)
+      GeneralUpdateMethodNamesDecorator.new(template_proxy)
+    end
+
+    def get_vm_proxy(vm, connection)
+      connection ||= ext_management_system.connect
+      vm_proxy = connection.get_resource_by_ems_ref(vm.ems_ref)
+      GeneralUpdateMethodNamesDecorator.new(vm_proxy)
+    end
+
+    def collect_disks_by_hrefs(disks)
+      vm_disks = []
+
+      ext_management_system.with_provider_connection do |rhevm|
+        disks.each do |disk|
+          begin
+            vm_disks << Ovirt::Disk.find_by_href(rhevm, disk)
+          rescue Ovirt::MissingResourceError
+            nil
+          end
+        end
+      end
+      vm_disks
+    end
+
+    def shutdown_guest(operation)
+      operation.with_provider_object(&:shutdown)
+      rescue Ovirt::VmIsNotRunning
+    end
+
+    def start_clone(source, clone_options, phase_context)
+      source.with_provider_object do |rhevm_template|
+        vm = rhevm_template.create_vm(clone_options)
+        populate_phase_context(phase_context, vm)
+      end
+    end
+
+    def vm_start(vm, cloud_init)
+      vm.with_provider_object do |rhevm_vm|
+        rhevm_vm.start { |action| action.use_cloud_init(true) if cloud_init }
+      end
+      rescue Ovirt::VmAlreadyRunning
+    end
+
+    def vm_stop(vm)
+      vm.with_provider_object(&:stop)
+      rescue Ovirt::VmIsNotRunning
+    end
+
+    def vm_suspend(vm)
+      vm.with_provider_object(&:suspend)
+    end
+
+    class NicsDecorator < SimpleDelegator
+      def name
+        self[:name]
+      end
+
+      def network
+        id = self[:network][:id]
+        OpenStruct.new(:id => id)
+      end
+    end
+
+    class GeneralUpdateMethodNamesDecorator < SimpleDelegator
```

Comment:
Ok, should I move all classes I created to separate files? They are all used only in this specific context as decorators.